SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -380,9 +387,15 @@ void USpatialNetDriver::CreateAndInitializeCoreClasses()
 	Dispatcher = NewObject<USpatialDispatcher>();
 	Sender = NewObject<USpatialSender>();
 	Receiver = NewObject<USpatialReceiver>();
-	GlobalStateManager = NewObject<UGlobalStateManager>();
+
+	if (Connection->GlobalStateManager == nullptr)
+	{
+		Connection->GlobalStateManager = NewObject<UGlobalStateManager>();
```

Comment:
Did we get to the bottom of why this doesn't cause the object free? If so, worth adding a comment.
Also probably worthwhile to initialise the staticcomponentview in the same way.

Reply:
Did not get to the bottom of this. Have added a comment and done the same with the ComponentView.